### PR TITLE
chore: support go 1.23

### PR DESCRIPTION
**Pull request by 42atomys**
Add support for go 1.23 as released https://go.dev/doc/go1.23

### CODE REVIEW

**Comment by 42atomys** [comment]
@microsoft-github-policy-service agree

**Comment by 42atomys** [comment]
@microsoft-github-policy-service agree

File: src/go/README.md [hunk 1]
```diff
@@ -24,18 +24,18 @@ You can directly reference pre-built versions of `Dockerfile` by using the `imag
...
 Refer to [this guide](https://containers.dev/guide/dockerfile) for more details.
 
 You can decide how often you want updates by referencing a [semantic version](https://semver.org/) of each image. For example:
 
-- `mcr.microsoft.com/devcontainers/go:1-1.22` (or `1-1.22-bookworm`, `1-1.22-bullseye`)
-- `mcr.microsoft.com/devcontainers/go:1.1-1.22` (or `1.1-1.22-bookworm`, `1.1-1.22-bullseye`)
-- `mcr.microsoft.com/devcontainers/go:1.1.9-1.22` (or `1.1.9-1.22-bookworm`, `1.1.9-1.22-bullseye`)
+- `mcr.microsoft.com/devcontainers/go:1-1.23` (or `1-1.23-bookworm`, `1-1.23-bullseye`)
+- `mcr.microsoft.com/devcontainers/go:1.1-1.23` (or `1.1-1.23-bookworm`, `1.1-1.23-bullseye`)
+- `mcr.microsoft.com/devcontainers/go:1.1.9-1.23` (or `1.1.9-1.23-bookworm`, `1.1.9-1.23-bullseye`)
```
Comment (samruddhikhandale) [inline src/go/README.md:36]:
```suggestion
- `mcr.microsoft.com/devcontainers/go:1-1.23` (or `1-1.23-bookworm`, `1-1.23-bullseye`)
- `mcr.microsoft.com/devcontainers/go:1.2-1.23` (or `1.2-1.23-bookworm`, `1.2-1.23-bullseye`)
- `mcr.microsoft.com/devcontainers/go:1.2.0-1.23` (or `1.2.0-1.23-bookworm`, `1.2.0-1.23-bullseye`)
```

File: src/go/manifest.json [hunk 2]
```diff
@@ -31,17 +41,27 @@
...
         "go:${VERSION}-1-bookworm",
         "go:${VERSION}-bookworm"
       ],
-			"1.22-bullseye": [
+      "1.23-bullseye": [
         "go:${VERSION}-1-bullseye",
         "go:${VERSION}-bullseye"
       ],
-			"1.21-bookworm": [
+      "1.22-bullseye": [
```
Comment (samruddhikhandale) [inline src/go/manifest.json:60]: Let's remove `1.22-bullseye` property, else there would be conflicts with dual pushes to same tag!

File: src/go/manifest.json [hunk 3]
```diff
@@ -13,15 +15,23 @@
       "go:${VERSION}-${VARIANT}"
```
Comment (samruddhikhandale) [inline src/go/manifest.json:15]: Can we update https://github.com/devcontainers/images/blob/403dc4ec7ef23de94e18d5a48566d8b971623017/src/go/manifest.json#L12 to use the newer version (ie 1.23-bookworm)? Thanks!

**Comment by samruddhikhandale** [comment]
Opened https://github.com/devcontainers/images/pull/1160 and https://github.com/devcontainers/images/issues/1162 for tracking this end-to-end.

File: src/go/manifest.json (hunk 3 above)
Comment (42atomys) [reply src/go/manifest.json:15]: Ah forgot this one good catch !

**Comment by 42atomys** [comment]
Thanks for the reviews and issues 🙏 Changes applied ✔️

**Review by samruddhikhandale — APPROVED** [review]
Great, looks good. Thanks! 

We'd release this new image on Monday.

**Pull request merged**